### ISSUE TITLE
Render block and follow messages

### ISSUE THREAD

**Issue by cinnamon-bun** [fraction/oasis#245]
**What's the problem you want solved?**

It's useful to know when a friend has blocked someone, in case you want to block them also.  

Sometimes a conversation happens in replies to the block message, so we want to be able to render block messages.

**Is there a solution you'd like to recommend?**

Show Block messages in the feed and allow replying to them.

Details
* Block messages can be private, so don't show your own private blocks as messages in any of the public feeds
* Don't show when YOU'VE been blocked?  (?)  You might still discover the conversation on the block message though.
* Don't show the user icon of the blocked feed, since it might be offensive
* If several people block the same feed, consolidate those into one message like Patchwork does?
* Call this "block" for now until we decide on renaming blocks #134 
* Which views should show block messages?  Most of them probably

**Follow and unfollow messages**

These are less important to show but still useful.
* Consolidate them like Patchwork does, since there are so many of them
* Ok to show user icons
* Should we show if someone has unfollowed you?

**Comment by christianbundy** [comment]
Yeah, being notified about blocks could be really useful. Where do you think we should put it?

I think I'd like to be able to see relationship changes, but I'm not sure whether I'd like to have them intermingled with posts from my friends. Do you think it's important to put this in pages like Latest, or could/should these start out in their own page (Relationships/Connections/etc) or something? I don't remember whether most social media sites put follow/unfollow/block messages in the main message feed, or whether it would be surprising for people. 

Alternatively, maybe blocks are something you get a notification for, so that you know that A Thing Happened and you have the opportunity to participate in a social immune system response? Or maybe that's toxic and feels bad.

Anyway: YUS. I'm a bit concerned about the Latest feed being blown up with follows, and I've really enjoyed not seeing them in Oasis, but there's utility in making sure that bad behavior is blocked ASAP so it doesn't spread to everyone.

**Comment by christianbundy** [comment]
Shower thought: maybe a page like [`follows.js`](https://oasis-demo.fraction.io/thread/%25OLPo%2FS1aR1X87rOE68vc6ucmJsV9A2jhoj7qoT8Pd9I%3D.sha256#%25OLPo%2FS1aR1X87rOE68vc6ucmJsV9A2jhoj7qoT8Pd9I%3D.sha256) that shows all of the info on relationships. If we could avoid a popularity contest, I think it might be useful to see a list of who to follow, who to block, or maybe (if the previous two don't solve the problem) a firehose of relationship changes?

**Comment by cinnamon-bun** [comment]
@christianbundy 

Yes a page for relationships!

Andre Staltz had interesting ideas for this:

> A new screen in Manyverse should list all accounts in the user's device, and their corresponding replication status. This could look a bit like a Torrent client when those feeds are being replicated & indexed.

> Beyond replication status, "My community" screen can also show hop distance, and total storage that this account incurs to the user's local disk. The replication status itself could actually be hidden unless there is a replication task currently in progress.

https://gitlab.com/staltz/manyverse/issues/745#ui-design-my-community

**Comment by cinnamon-bun** [comment]
While I'm making wishes, I'll wish for a separate list of pubs and your relationship with them.  I don't know how to tell if a peer is a pub or not.

**Comment by georgeowell** [comment]
Yep, I think a separate view for relationships. (un)Blocks, (un)follows,

It could even show who is liking what perhaps?
